chain/jsonrpc/src/lib.rs

```diff
@@ -621,13 +627,11 @@ impl JsonRpcHandler {
             .map_err(|err| err.into())))
     }
 
-    async fn block(&self, params: Option<Value>) -> Result<Value, RpcError> {
-        let block_reference = if let Ok((block_id,)) = parse_params::<(BlockId,)>(params.clone()) {
-            BlockReference::BlockId(block_id)
-        } else {
-            parse_params::<BlockReference>(params)?
-        };
-        jsonify(self.view_client_addr.send(GetBlock(block_reference)).await)
+    async fn block(
+        &self,
+        request_data: near_primitives::rpc::RpcBlockRequest,
+    ) -> Result<BlockView, RpcBlockError> {
```

Comment:
And the final step would be to use `RpcBlockResponse`, which will wrap `BlockView` (flatten). :pray: